Source/Classes/SLKTypingIndicatorProtocol.h

```diff
@@ -0,0 +1,44 @@
+//
+//   Copyright 2014 Slack Technologies, Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+/** Generic protocol needed when customizing your own typing indicator view. */
+@protocol SLKTypingIndicatorProtocol <NSObject>
+@required
+
+/**
+ Returns YES if the indicator is visible.
+ SLKTextViewController depends on this property internally, by observing its value changes to update the typing indicator view's constraints automatically.
+ You can simply @synthesize this property to make it KVO compliant, or override its setter method and wrap its implementation with -willChangeValueForKey: and -didChangeValueForKey: methods, for more complex KVO compliance.
+ */
+@property (nonatomic, getter = isVisible) BOOL visible;
```

Comment:
In order to satisfy this requirement in a Swift implementation, I had to use the `@objc()` attribution to get a different name for the setter and the getter. Also, since I only wanted this for the getter, and not the setter, I had to actually specify a setter and getter implementation. Which again did not auto-synthesize a backing variable. Therefore, I had to create a new, private, backing variable to store it. Also, to get KVO working, I had to explicitly mark it for dynamic dispatch. In conclusion, this satisfies the protocol requirements for a Swift class:

```swift
class MyTypingIndicatorView: UIView, SLKTypingIndicatorProtocol {

    private var _visible: Bool = false
    dynamic var visible: Bool {
        @objc(isVisible) get {
            return _visible
        }
        set {
            _visible = newValue
        }
    }

}
```

Then setting `visible = true` or `visible = false` anywhere in your code works magically.